pkg/controller/daemon/daemoncontroller.go

```diff
@@ -716,6 +744,17 @@ func (dsc *DaemonSetsController) syncDaemonSet(key string) error {
 		}
 	}
 
+	dsNeedsSync = dsc.expectations.SatisfiedExpectations(dsKey)
```

Comment:
`manage` can set new expectations. I want to run updates only if `manage` didn't change anything.